### ISSUE TITLE
Install Fail: db.properties': No such file or directory

### ISSUE THREAD

**Issue by tampler** [apache/cloudstack#5226]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Installation
~~~

##### CLOUDSTACK VERSION

~~~
4.15.1.0.focal
~~~

##### OS / ENVIRONMENT

Ubuntu 20.04.2
Linux 5.8.0-59-generic

##### SUMMARY

Installation fails for a non-root user.

CloudStack Management server fails during installation due to TWO errors:

1. mysql-server-8.0 (8.0.25-0ubuntu0.20.04.1) won't create an included folder `/etc/mysql/conf.d` which yields in a config failure during installation
2. CloudStack won't create a file :
```
Setting up cloudstack-management (4.15.1.0~focal) ...
usermod: no changes
chmod: cannot access '/etc/cloudstack/management/db.properties': No such file or directory
dpkg: error processing package cloudstack-management (--configure):
 installed cloudstack-management package post-installation script subprocess returned error exit status 1
Setting up mysql-server (8.0.25-0ubuntu0.20.04.1) ...
Errors were encountered while processing:
 cloudstack-management
 ```

##### STEPS TO REPRODUCE

~~~
apt-get install cloudstack-management cloudstack-usage
~~~

##### EXPECTED RESULTS

~~~
Installation succeeds 
~~~

##### ACTUAL RESULTS

Installation failed and requires a manual intervention

~~~

~~~

**Comment by DaanHoogland** [comment]
@tampler why do you install as a non-root user? I think that is your issue.

**Comment by tampler** [comment]
Hello. Yes, I'm installing as a sudo user, but NOT a root. The reason is that I can't reset a root password and can't temporarily access the hardware.

I found a possible reason. I have Python 3.8.10 shipped with Ubuntu 20.04.2
I'm getting the following errors during the config stage:
```
Setting up python3-mysql.connector (8.0.15-2build1) ...                                                                                                                      
/usr/lib/python3/dist-packages/mysqlx/expr.py:972: SyntaxWarning: "is" with a literal. Did you mean "=="?                                                                    
  return "({0})".format(dimension[0]) if len(dimension) is 1 else \                                                                                                          
/usr/lib/python3/dist-packages/mysqlx/statement.py:122: SyntaxWarning: "is" with a literal. Did you mean "=="?                                                               
  return (default_schema if len(temp) is 1 else temp[0].strip(quote),    
...
Setting up python3-netaddr (0.7.19-3) ...
/usr/lib/python3/dist-packages/netaddr/strategy/__init__.py:189: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if word_sep is not '':
```

and all this yields in the following
```
chmod: cannot access '/etc/cloudstack/management/db.properties': No such file or directory
dpkg: error processing package cloudstack-management (--configure):
 installed cloudstack-management package post-installation script subprocess returned error exit status 1
```

**Comment by weizhouapache** [comment]
> Hello @DaanHoogland . Yes, I'm installing as a sudo user, but NOT a root. The reason is that I can't reset a root password and can't temporarily access the hardware.
> 
> I found a possible reason. I have Python 3.8.10 shipped with Ubuntu 20.04.2
> I'm getting the following errors during the config stage:
> 
> ```
> Setting up python3-mysql.connector (8.0.15-2build1) ...                                                                                                                      
> /usr/lib/python3/dist-packages/mysqlx/expr.py:972: SyntaxWarning: "is" with a literal. Did you mean "=="?                                                                    
>   return "({0})".format(dimension[0]) if len(dimension) is 1 else \                                                                                                          
> /usr/lib/python3/dist-packages/mysqlx/statement.py:122: SyntaxWarning: "is" with a literal. Did you mean "=="?                                                               
>   return (default_schema if len(temp) is 1 else temp[0].strip(quote),    
> ...
> Setting up python3-netaddr (0.7.19-3) ...
> /usr/lib/python3/dist-packages/netaddr/strategy/__init__.py:189: SyntaxWarning: "is not" with a literal. Did you mean "!="?
>   if word_sep is not '':
> ```
> 
> and all this yields in the following
> 
> ```
> chmod: cannot access '/etc/cloudstack/management/db.properties': No such file or directory
> dpkg: error processing package cloudstack-management (--configure):
>  installed cloudstack-management package post-installation script subprocess returned error exit status 1
> ```

@tampler I think you can ignore these warnings.
can you run "sudo -i" and then install packages ?

**Comment by tampler** [comment]
@ustcweizhou Thanks for your advice ! It saved my day. I was able to seamlessly install and setup CloudStack under the root account. 

However, I'm still facing the issue #5051 adding a KVM host and I'd appreciate your help and advice on that topic.

Closing this issue. Thanks everyone for your help !
